userspace/falco/webserver.cpp

```diff
@@ -14,258 +14,100 @@ See the License for the specific language governing permissions and
-{
-	m_config = config;
-	m_engine = engine;
-	m_outputs = outputs;
-	m_k8s_audit_event_source_idx = k8s_audit_event_source_idx;
-}
-
-template<typename T, typename... Args>
-std::unique_ptr<T> make_unique(Args &&... args)
-{
-	return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+    stop();
 }
 
-void falco_webserver::start()
+void falco_webserver::start(
+		uint32_t listen_port,
+		std::string& healthz_endpoint,
+		std::string &ssl_certificate,
+		bool ssl_enabled)
 {
-	if(m_server)
-	{
-		stop();
-	}
-
-	if(!m_config)
-	{
-		throw falco_exception("No config provided to webserver");
-	}
-
-	if(!m_engine)
-	{
-		throw falco_exception("No engine provided to webserver");
-	}
-
-	if(!m_outputs)
-	{
-		throw falco_exception("No outputs provided to webserver");
-	}
-
-	std::vector<std::string> cpp_options = {
-		"num_threads", to_string(1)};
-
-	if(m_config->m_webserver_ssl_enabled)
-	{
-		cpp_options.push_back("listening_ports");
-		cpp_options.push_back(to_string(m_config->m_webserver_listen_port) + "s");
-		cpp_options.push_back("ssl_certificate");
-		cpp_options.push_back(m_config->m_webserver_ssl_certificate);
-	}
-	else
-	{
-		cpp_options.push_back("listening_ports");
-		cpp_options.push_back(to_string(m_config->m_webserver_listen_port));
-	}
-
-	try
-	{
-		m_server = make_unique<CivetServer>(cpp_options);
-	}
-	catch(CivetException &e)
-	{
-		throw falco_exception(std::string("Could not create embedded webserver: ") + e.what());
-	}
-	if(!m_server->getContext())
-	{
-		throw falco_exception("Could not create embedded webserver");
-	}
-
-	m_k8s_audit_handler = make_unique<k8s_audit_handler>(m_engine, m_outputs, m_k8s_audit_event_source_idx);
-	m_server->addHandler(m_config->m_webserver_k8s_audit_endpoint, *m_k8s_audit_handler);
-	m_k8s_healthz_handler = make_unique<k8s_healthz_handler>();
-	m_server->addHandler(m_config->m_webserver_k8s_healthz_endpoint, *m_k8s_healthz_handler);
+    if (m_running)
+    {
+        throw falco_exception(
+            "attempted restarting webserver without stopping it first");
+    }
+
+    // allocate and configure server
+    if (ssl_enabled)
+    {
+        m_server = new httplib::SSLServer(
+            ssl_certificate.c_str(),
+            ssl_certificate.c_str());
+    }
+    else
+    {
+        m_server = new httplib::Server();
+    }
+
+    // setup healthz endpoint
+    m_server->Get(healthz_endpoint,
+        [](const httplib::Request &, httplib::Response &res) {
+            res.set_content("{\"status\": \"ok\"}", "application/json");
+        });
+
+    // run server in a separate thread
+    if (!m_server->is_valid())
+    {
+        delete m_server;
+        m_server = NULL;
+        throw falco_exception("invalid webserver configuration");
+    }
+
+    std::atomic<bool> failed;
+    failed.store(false, std::memory_order_relaxed);
+    m_server_thread = std::thread([this, listen_port, &failed]
+    {
+        try
+        {
+            this->m_server->listen("localhost", listen_port);
```

Comment:
```suggestion
            this->m_server->listen("0.0.0.0", listen_port);
```
I guess it should be bound to `0.0.0.0` and not just `127.0.0.1`.

Reply:
PS
It would be better to let the user choose, but this PR is already enough big.